### PR TITLE
ci, lint: run more often, lint: switch to eslint-config-treesitter, ci, publish: update to `macos-latest` and test generating

### PR DESCRIPTION
**Pull request by sebastiaanspeck**
## Summary by CodeRabbit

- **New Features**
	- CI workflow now triggers on any branch or path for push and pull request events, enhancing flexibility.
	- Introduced a new JSON configuration file for the Tree-sitter parser defining the grammar for the Magik programming language, including metadata and supported bindings.
	- New Go module file created for the Magik project, specifying dependencies and Go version.
	- Added a new field in node-types.json to recognize source_file as a root node.

- **Bug Fixes**
	- Removed restrictive filters that previously limited workflow triggers, ensuring more comprehensive testing and linting.

- **Chores**
	- Updated dependencies in package.json, replacing eslint-config-google with eslint-config-treesitter.
	- Added a new script for prebuildify in package.json and modified the test script for consistency.
	- Introduced a new schema declaration in grammar.json for validation.
	- Removed the entire eslintConfig section from package.json to simplify linting configuration.
	- Enhanced clarity in allocation function signatures in alloc.h.
	- Removed logging function pointer from the TSLexer struct in parser.h.

### CODE REVIEW

**Comment by sebastiaanspeck** [comment]
The reason for this PR is because the dependabot-PRs did not trigger the (failing) workflows. In this PR I will fix the failing workflows.

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
   },
   "devDependencies": {
     "eslint": "^9.12.0",
-    "eslint-config-google": "^0.14.0",
+    "eslint-config-treesitter": "^1.0.2",
```
[the remark this answers is not in the document]
Comment (sebastiaanspeck) [reply package.json:41]: We added `.eslint.config.mjs` which is also valid.

File: tree-sitter.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "authors": [
+      {
+        "name": "Robin Putters",
+        "email": "robinputters@github.com"
+      }
+    ],
+    "links": {
+      "repository": "https://github.com/krn-robin/tree-sitter-magik"
+    }
+  },
```
[the remark this answers is not in the document]
Comment (sebastiaanspeck) [reply tree-sitter.json]: See my last commit :smile:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -47,6 +34,11 @@ jobs:
         uses: tree-sitter/parser-test-action@v2
         with:
           test-rust: ${{runner.os == 'Linux'}}
+      - name: Check unchanged parser
+        shell: bash
+        run: |
+          git status
+          exit $(git status --porcelain=v1 2>/dev/null | grep -c src/parser.c)
```
[the remark this answers is not in the document]
Comment (sebastiaanspeck) [reply .github/workflows/ci.yml]: This has been removed in a next commit. Please re-check.

**Review by krn-robin — DISMISSED** [review]

**Review by krn-robin — APPROVED** [review]

**Pull request merged**